### PR TITLE
update dependencies to newer versions for sprawl compatibility

### PR DESCRIPTION
**Pull request by zook-unity**
# Peer Review Information

upgraded `google-cloud-storage` to >=1.31.1 and `numpy` to >=1.18.2

# Pull Request Check List

no new tests or documentation.

### CODE REVIEW

**Comment by zook-unity** [comment]
one note: i have not done any additional testing for cloud training and storage, &c.

File: datasetinsights/evaluation_metrics/average_recall_2d.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ class MeanAverageRecallAverageOverIOU(EvaluationMetric):
     TYPE = "scalar"
 
     IOU_THRESHOULDS = np.linspace(
-        0.5, 0.95, np.round((0.95 - 0.5) / 0.05) + 1, endpoint=True
+        0.5, 0.95, int(np.round((0.95 - 0.5) / 0.05)) + 1, endpoint=True
```
Comment (adason) [inline datasetinsights/evaluation_metrics/average_recall_2d.py:114]: Is this the issue (feature) that was fixed (introduced) in numpy 1.18+?

File: pyproject.toml [hunk 2]
```diff
@@ -26,10 +26,10 @@ include = [
 [tool.poetry.dependencies]
 python = "^3.7"
 cython = "^0.29.14"
-google-cloud-storage = ">=1.24.1 <=1.28.1"
+google-cloud-storage = ">=1.31.1"
```
Comment (adason) [inline pyproject.toml]: I am assuming this [issue](https://github.com/Unity-Technologies/datasetinsights/pull/82) was fixed in `google-cloud-storage 1.31+?

File: pyproject.toml [hunk 3]
```diff
@@ -26,10 +26,10 @@ include = [
...
 jupyter = "^1.0.0"
 kornia = "^0.1.4"
-numpy = ">=1.17 < 1.18"
+numpy = ">=1.18.2"
```
Comment (adason) [inline pyproject.toml]: Let's use `^1.18.2` or `>=1.18.2 <2.0.0` (they are equivalent syntex). This prevents unexpected major version bump.

File: datasetinsights/evaluation_metrics/average_recall_2d.py (hunk 1 above)
Comment (zook-unity) [reply datasetinsights/evaluation_metrics/average_recall_2d.py:114]: yup. linspace expects an int but round returns a numpy float64. i'm guessing that used to get converted to an int before, but it's probably not a good practice to rely on implicit type conversion either way.

File: pyproject.toml (hunk 2 above)
Comment (zook-unity) [reply pyproject.toml]: i did not know about that issue. it looks like 1.31.1 merged those changes. we could do a later version if that matters, but i picked this to align with what sprawl uses currently

release notes for the fix: https://github.com/googleapis/python-storage/releases/tag/v1.31.1
`blob: base64 includes additional characters`

File: pyproject.toml (hunk 3 above)
Comment (zook-unity) [reply pyproject.toml]: ah, good call! making that update.

File: pyproject.toml (hunk 2 above)
Comment (adason) [reply pyproject.toml]: Good to know the issue is fixed. Let's also include `<2.0.0` in the requirements to make sure we don't accidentally bump the major version.

Comment (zook-unity) [reply pyproject.toml]: updated. anything look off?

**Review by adason — APPROVED** [review]

**Pull request merged**